src/inference/tests/functional/get_supported_property_test.cpp

```diff
@@ -0,0 +1,176 @@
+                                                                std::string("mock_engine") + IE_BUILD_POSTFIX),
+                             m_plugin_name);
+        m_mock_plugin = plugin;
+    }
+
+    void TearDown() override {
+        core.unload_plugin(m_plugin_name);
+    }
+
+protected:
+    std::shared_ptr<void> m_so;
+    std::shared_ptr<InferenceEngine::IInferencePlugin> m_mock_plugin;
+    ov::AnyMap m_properties;
+    ov::AnyMap m_expected_properties;
+    std::string m_plugin_name{"MOCK_HARDWARE"};
+};
+
+static std::string getTestCaseName(const testing::TestParamInfo<TestParam>& obj) {
+    ov::AnyMap properties = std::get<0>(obj.param);
+    ov::AnyMap expected_properties = std::get<1>(obj.param);
+    std::ostringstream result;
+    result << "query_property_";
+    for (auto& item : properties) {
+        result << item.first << "_" << item.second.as<std::string>() << "_";
+    }
+    result << "expected_property_";
+    for (auto& item : expected_properties) {
+        result << item.first << "_" << item.second.as<std::string>() << "_";
+    }
+    auto name = result.str();
+    name.pop_back();
+    return name;
+}
+
+TEST_P(GetPropertyTest, canGenerateCorrectPropertyList) {
+    auto plugin = std::make_shared<MockHardWarePlugin>();
+    std::shared_ptr<InferenceEngine::IInferencePlugin> base_plugin = plugin;
+    reg_plugin(core, base_plugin);
+    core.get_property(m_plugin_name, ov::supported_properties);
+    std::map<std::string, std::string> config;
+    for (auto&& value : m_properties) {
+        config.emplace(value.first, value.second.as<std::string>());
+    }
+    auto actual_output = m_mock_plugin->GetCore()->GetSupportedConfig(m_plugin_name, config);
+    for (auto& iter : m_expected_properties) {
+        ASSERT_TRUE(actual_output.find(iter.first) != actual_output.end());
+        ASSERT_EQ(actual_output.find(iter.first)->second, iter.second.as<std::string>());
+    }
+    for (auto& iter : m_properties) {
+        if (m_expected_properties.find(iter.first) == m_expected_properties.end()) {
+            ASSERT_TRUE(actual_output.find(iter.first) == actual_output.end());
+        }
+    }
+}
+
+static const std::vector<TestParam> test_variants = {
+    TestParam{ov::AnyMap({ov::hint::allow_auto_batching(false), ov::num_streams(2)}),
+              ov::AnyMap({ov::hint::allow_auto_batching(false), ov::num_streams(2)})},
+    TestParam{ov::AnyMap({ov::auto_batch_timeout(0), ov::enable_profiling(false)}),
+              ov::AnyMap({ov::auto_batch_timeout(0)})},
+    TestParam{ov::AnyMap({ov::cache_dir("test"), ov::force_tbb_terminate(false)}),
+              ov::AnyMap({ov::cache_dir("test"), ov::force_tbb_terminate(false)})},
+    TestParam{ov::AnyMap({ov::cache_dir("test"),
+                          ov::device::properties("MOCK_HARDWARE", ov::num_streams(2), ov::enable_profiling(true))}),
+              ov::AnyMap({ov::cache_dir("test"), ov::num_streams(2)})},
+};
```

Comment:
should we add cases how the whole story works via HETERO / AUTO ?  E.g. when `GetSupportedConfig` is called for virtual plugins, then properties are kept.